app/controllers/support/providers/accredited_partnerships_controller.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+module Support
+  module Providers
+    class AccreditedPartnershipsController < ApplicationController
+      include ClearStashable
+
+      helper_method :accredited_provider_id
+
+      before_action :reset_accredited_provider_form, only: %i[index]
+
+      def index
+        @pagy, @partnerships = pagy(provider.accredited_partnerships)
+      end
+
+      def new
+        partnership = provider.accredited_partnerships.build
+        partnership.assign_attributes(accredited_provider_id: params[:accredited_provider_id] || ProviderPartnershipForm.new(current_user, provider).accredited_provider_id)
+
+        @accredited_provider_form = ProviderPartnershipForm.new(current_user, partnership)
+        return unless partnership.invalid?
+
+        redirect_to search_support_recruitment_cycle_provider_accredited_providers_path(RecruitmentCycle.current.year), flash: { error: { message: "#{Provider.find(params[:accredited_provider_id]).name_and_code} partnership already exists" } }
+      end
+
+      def edit
+        partnership = provider.accredited_partnerships.find_by(accredited_provider: partner)
+        params = { accredited_provider_id: partner.id, description: partnership.description }
+        @accredited_provider_form = ::ProviderPartnershipForm.new(current_user, partnership, params:)
+      end
+
+      def create
+        @accredited_provider_form = ::ProviderPartnershipForm.new(current_user, provider, params: accredited_provider_params)
+
+        if @accredited_provider_form.stash
+          redirect_to check_support_recruitment_cycle_provider_accredited_partnerships_path(accredited_provider_id: partnership_params[:accredited_provider_id])
+        else
+          render :new
+        end
+      end
+
+      def update
+        partnership = provider.accredited_partnerships.find_by(accredited_provider: partner)
+        @accredited_provider_form = ::ProviderPartnershipForm.new(current_user, partnership, params: accredited_provider_params)
+
+        if @accredited_provider_form.save!
+          redirect_to support_recruitment_cycle_provider_accredited_partnerships_path(
+            recruitment_cycle_year: @recruitment_cycle.year,
+            provider_id: @provider.id
+          )
+
+          flash[:success] = t('support.providers.accredited_providers.edit.updated')
```

Comment:
Let's add the flash to the redirection.

```suggestion
          redirect_to support_recruitment_cycle_provider_accredited_partnerships_path(
            recruitment_cycle_year: @recruitment_cycle.year,
            provider_id: @provider.id
          ), flash: { success: t('support.providers.accredited_providers.edit.updated') }
```